requests/adapters.py

```diff
@@ -493,13 +493,12 @@ def send(self, request, stream=False, timeout=None, verify=True, cert=None, prox
                 # TODO: Remove this in 3.0.0: see #2811
                 if not isinstance(e.reason, NewConnectionError):
                     raise ConnectTimeout(e, request=request)
-
-            if isinstance(e.reason, ResponseError):
+            elif isinstance(e.reason, ResponseError):
```

Comment:
There have been a couple issues opened where users of requests have seen that this clustering of multiple error messages (joining the initial error message alongside the max requests retried) can be confusing to the end user, because it isn't abundantly clear with error message is causing the problem.  Can we come up with a proposed solution for that?  That was my initial intent while writing this PR.